### PR TITLE
[ticket/16114] mcp_topic_post_author_full_{append/prepend}

### PR DESCRIPTION
**Pull request by dark-1**
PHPBB3-16114

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16114

### CODE REVIEW

File: phpBB/styles/prosilver/template/mcp_topic.html [hunk 1]
```diff
@@ -120,7 +120,7 @@ <h3><a href="{postrow.U_POST_DETAILS}">{postrow.POST_SUBJECT}</a></h3>
 				<p class="author">
 					<a href="#pr{postrow.POST_ID}" title="{postrow.MINI_POST}">
 						<i class="icon fa-file fa-fw icon-lightgray icon-tiny" aria-hidden="true"></i><span class="sr-only">{postrow.MINI_POST}</span>
-					</a> {L_POSTED} {postrow.POST_DATE} {L_POST_BY_AUTHOR} <strong>{postrow.POST_AUTHOR_FULL}</strong><!-- IF postrow.U_MCP_DETAILS --> [ <a href="{postrow.U_MCP_DETAILS}">{L_POST_DETAILS}</a> ]<!-- ENDIF -->
+					</a> {L_POSTED} {postrow.POST_DATE} {L_POST_BY_AUTHOR} <!-- EVENT mcp_topic_postrow_post_details_author_username_prepend --><strong>{postrow.POST_AUTHOR_FULL}</strong><!-- EVENT mcp_topic_postrow_post_details_author_username_append --><!-- IF postrow.U_MCP_DETAILS --> [ <a href="{postrow.U_MCP_DETAILS}">{L_POST_DETAILS}</a> ]<!-- ENDIF -->
```
Comment (3D-I) [inline phpBB/styles/prosilver/template/mcp_topic.html]: Use Twig for template events, as reccomended.

**Comment by dark-1** [comment]
!set Event

File: phpBB/styles/prosilver/template/mcp_topic.html (hunk 1 above)
Comment (dark-1) [reply phpBB/styles/prosilver/template/mcp_topic.html]: Done !!! Thanks @3D-I  👍

**Comment by 3D-I** [comment]
Events' name are too long and redundant IMHO.

**Comment by dark-1** [comment]
> Events' name are too long and redundant IMHO.

So what I have to do ??? any help please... @3D-I ???

File: phpBB/styles/prosilver/template/mcp_topic.html [hunk 2]
```diff
@@ -120,7 +120,7 @@ <h3><a href="{postrow.U_POST_DETAILS}">{postrow.POST_SUBJECT}</a></h3>
 				<p class="author">
 					<a href="#pr{postrow.POST_ID}" title="{postrow.MINI_POST}">
 						<i class="icon fa-file fa-fw icon-lightgray icon-tiny" aria-hidden="true"></i><span class="sr-only">{postrow.MINI_POST}</span>
-					</a> {L_POSTED} {postrow.POST_DATE} {L_POST_BY_AUTHOR} <!-- EVENT mcp_topic_postrow_post_details_author_username_prepend --><strong>{postrow.POST_AUTHOR_FULL}</strong><!-- EVENT mcp_topic_postrow_post_details_author_username_append --><!-- IF postrow.U_MCP_DETAILS --> [ <a href="{postrow.U_MCP_DETAILS}">{L_POST_DETAILS}</a> ]<!-- ENDIF -->
+					</a> {L_POSTED} {postrow.POST_DATE} {L_POST_BY_AUTHOR} {% EVENT mcp_topic_postrow_post_details_author_username_prepend %}<strong>{postrow.POST_AUTHOR_FULL}</strong>{% EVENT mcp_topic_postrow_post_details_author_username_append %}<!-- IF postrow.U_MCP_DETAILS --> [ <a href="{postrow.U_MCP_DETAILS}">{L_POST_DETAILS}</a> ]<!-- ENDIF -->
```
Comment (3D-I) [inline phpBB/styles/prosilver/template/mcp_topic.html]: ´mcp_topic_post_author_full_prepend´ ?

Comment (dark-1) [reply phpBB/styles/prosilver/template/mcp_topic.html]: What about `mcp_topic_postrow_author_full_prepend` ???

Comment (3D-I) [reply phpBB/styles/prosilver/template/mcp_topic.html]: Whý? That's a part of the redundancies, there can be only one _postrow_ in there. ;)

Comment (dark-1) [reply phpBB/styles/prosilver/template/mcp_topic.html]: Yea!! I just replaced your `_post_` with `_postrow_` ... to make it more close to `mcp_topic_postrow_post_details_before` ... & it has only one `postrow`...

**Comment by 3D-I** [comment]
Have a look at how the other events in this file are named, like

`<!-- EVENT mcp_topic_topic_title_before -->`

**Comment by dark-1** [comment]
> Have a look at how the other events in this file are named, like
> 
> `<!-- EVENT mcp_topic_topic_title_before -->`

Hmm!! let's ask @marc1706 , what we should do ???

**Comment by 3D-I** [comment]
Look, we are in the ´mcp_topic´ html file and there is only a ´postrow´ inside of it... got it?
So it is understood that we are referring to it, it is useless to make it known in vain but only to lengthen the name of an event.

**Comment by marc1706** [comment]
Yes, please make the event name shorter as @3D-I suggested

**Comment by dark-1** [comment]
Ok, give me some time...

**Comment by dark-1** [comment]
@marc1706 , It's done 👍

**Comment by mrgoldy** [comment]
I feel like the `_full_` is redundant aswell.
And is there any guidelines on when to use `before/after` and `prepend/append`?
Cause I feel like append and prepend should be used when it's inside the actual tag, for example `{% EVENT title_before %}<h3>{% EVENT title_prepend %}My awesome title...`.
Which would make this event an `before/after` and not `append/prepend`.

**Comment by dark-1** [comment]
@mrgoldy ,
Nah!!! `_full_` is required...
Check here for guidelines on when to use `before/after` and `prepend/append` :
http://area51.phpbb.com/docs/32x/coding-guidelines.html#template-events

Best regards 👍

**Comment by 3D-I** [comment]
Well, the `_FULL` basically it is also redundant, since there is not a `POST_AUTHOR` of any sort. :)

By my side though is ok.

**Comment by dark-1** [comment]
Now Somebody got to give me the final String please...

**Comment by 3D-I** [comment]
> By my side though is ok.

**Comment by dark-1** [comment]
@marc1706 , should I make it shorter or keep it as how it's now... ???

**Comment by marc1706** [comment]
IMHO fine as that variable is called POST_AUTHOR_FULL.

**Pull request merged**